analyzers/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/InitializationVectorShouldBeRandom.cs

```diff
@@ -0,0 +1,206 @@
+                    AssignmentExpressionSyntax assignment => AssignmentExpressionPostProcess(assignment, programState),
+                    _ => programState
+                };
+
+            private ProgramState ObjectCreationPostProcess(ObjectCreationExpressionSyntax objectCreation, ProgramState programState)
+            {
+                if (semanticModel.GetSymbolInfo(objectCreation).Symbol is {} symbol
+                    && symbol.ContainingType is {} symbolType
+                    && symbolType.DerivesFrom(KnownType.System_Security_Cryptography_SymmetricAlgorithm))
+                {
+                    var symbolicValue = programState.PeekValue();
+                    programState = programState.SetConstraint(symbolicValue, CryptographyIVSymbolicValueConstraint.NotInitialized);
+                }
+
+                return programState;
+            }
+
+            private ProgramState InvocationExpressionPostProcess(InvocationExpressionSyntax invocation, ProgramState programState)
+            {
+                if (IsCreateMethod(invocation, semanticModel))
+                {
+                    var symbolicValue = programState.PeekValue();
+                    programState = programState.SetConstraint(symbolicValue, CryptographyIVSymbolicValueConstraint.NotInitialized);
+                }
+                else if (IsGenerateIVMethod(invocation, semanticModel))
+                {
+                    var symbolicValue = GetSymbolicValue(invocation, programState);
+                    programState = programState.SetConstraint(symbolicValue, CryptographyIVSymbolicValueConstraint.Initialized);
+                }
+                else if (IsSanitizer(invocation, semanticModel)
+                         && semanticModel.GetSymbolInfo(invocation.ArgumentList.Arguments[0].Expression).Symbol is {} symbol
+                         && symbol.GetSymbolType().Is(KnownType.System_Byte_Array)
+                         && symbol.HasConstraint(ByteArraySymbolicValueConstraint.Constant, programState))
+                {
+                    var symbolicValue = programState.GetSymbolValue(symbol);
+                    programState = programState.SetConstraint(symbolicValue, ByteArraySymbolicValueConstraint.Modified);
+                }
+
+                return programState;
+            }
+
+            private ProgramState AssignmentExpressionPostProcess(AssignmentExpressionSyntax assignment, ProgramState programState) =>
+                assignment.Left is MemberAccessExpressionSyntax memberAccess
+                && IsIVMemberAccess(memberAccess)
+                && semanticModel.GetSymbolInfo(memberAccess.Expression).Symbol is {} leftSymbol
+                && programState.GetSymbolValue(leftSymbol) is {} leftSymbolicValue
+                && semanticModel.GetSymbolInfo(assignment.Right).Symbol is {} rightSymbol
+                && programState.GetSymbolValue(rightSymbol) is {} rightSymbolicValue
```

Comment:
Improvement idea:
These 4 lines seems repetitive && the intermediate `leftSymbol`/`rightSymbol` state is not needed outside the 2 line scope && there's a lot of `&&` in the condition. Individually, it's fine, together, we might improve it by extracting it to some 
```
&& GetSymbolicValue(programState, memberAccess.Expression) is {} leftSymbolicValue
&& GetSymbolicValue(programState, assignment.Right) is {} rightSymbolicValue
```

Reply:
..aaand there's already `GetSymbolicValue` method that could call this one as well.